src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/component-manifest.targets

```diff
@@ -0,0 +1,94 @@
+      <_MonoRuntimeComponentCurrentLinking>@(_MonoRuntimeComponentLinking->WithMetadataValue('RuntimeIdentifier', '$(RuntimeIdentifier)'))</_MonoRuntimeComponentCurrentLinking>
+    </PropertyGroup>
+    <ItemGroup Condition="'$(_MonoRuntimeComponentCurrentLinking)' == 'dynamic'">
+      <_MonoRuntimeComponentName Include="@(_MonoRuntimeAvailableComponents)" Condition="'%(RuntimeIdentifier)' == '$(RuntimeIdentifier)'">
+        <ComponentLib>libmono-component-%(Identity)$(_MonoRuntimeComponentCurrentSharedLibExt)</ComponentLib>
+      </_MonoRuntimeComponentName>
+    </ItemGroup>
+    <ItemGroup Condition="'$(_MonoRuntimeComponentCurrentLinking)' == 'static'">
+      <_MonoRuntimeComponentName Include="@(_MonoRuntimeAvailableComponents)" Condition="'%(RuntimeIdentifier)' == '$(RuntimeIdentifier)'">
+        <ComponentLib>libmono-component-%(Identity)-static$(_MonoRuntimeComponentCurrentStaticLibExt)</ComponentLib>
+        <ComponentStubLib>libmono-component-%(Identity)-stub-static$(_MonoRuntimeComponentCurrentStaticLibExt)</ComponentStubLib>
+      </_MonoRuntimeComponentName>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_MonoSelectRuntimeComponents" DependsOnTargets="_MonoComputeAvailableComponentDefinitions">
+    <!-- Input: _MonoComponent item list of names of components that will be enabled (to be defined
+         by the workload) -->
+    <!-- Output: _MonoRuntimeSelectedComponents subset of _MonoRuntimeComponentName items that were
+         in _MonoComponent -->
+    <!-- Output: _MonoRuntimeSelectedStubComponents (static linking only) subset of
+         _MonoRuntimeComponentName items that were not in _MonoComponent -->
+    <!-- Output: _MonoRuntimeUnSelectedComponents subset of _MonoRuntimeComponentName items that were not in _MonoComponent. That is, the components that are available but that weren't selected. -->
+    <!-- Output: _MonoRuntimeUnSelectedStubComponents (static linking only) subset of _MonoRuntimeComponentName items that were in _MonoComponent.  That is, components that were selected - so their stubs should not be linked. -->
+    <!-- Output: _MonoRuntimeComponentLink item list of the filenames to link (relative paths
+         relative to the native/ directory of the runtime pack). For dynamic linking, this is only
+         the enabled components.  For static linking, it also includes stubs for the non-selected
+         components -->
+    <!-- Output: _MonoRuntimeComponentDontLink item list of the filename to exclude from linking
+         (relative paths relative to the native/ directory of the runtime pack). For static linking,
+         this is stubs or components that must not be passed to the native linker. For dynamic
+         linking, this is copmonent shared libraries that must not be bundled with the app
+         -->
```

Comment:
Should these two be combined into one item, and differentiated by some metadata?
And same question about `_MonoRuntime*Selected*Components`.

Reply:
No, they're completely unrelated.  One is for workloads that want to maintain a whitelist of stuff from `native/` that they will use.  The other is for workloads that just want to grab `native/*.a` and want an exclude list to tell them what to leave out.